### PR TITLE
Fix golint error for pkg/kubelet/types

### PR DESCRIPTION
**Pull request by qw1mb0**
What type of PR is this?
/kind documentation

What this PR does / why we need it:
- Fix lint errors for the whole directory `pkg/kubelet/types`
- Remove `pkg/kubelet/types` from `hack/.golint_failures`;
- Fix some typo from comments.

Which issue(s) this PR fixes: https://github.com/kubernetes/kubernetes/issues/68026

Special notes for your reviewer:

Does this PR introduce a user-facing change?:
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/types/labels.go [hunk 1]
```diff
@@ -17,24 +17,32 @@ limitations under the License.
 package types
 
 const (
-	KubernetesPodNameLabel       = "io.kubernetes.pod.name"
-	KubernetesPodNamespaceLabel  = "io.kubernetes.pod.namespace"
-	KubernetesPodUIDLabel        = "io.kubernetes.pod.uid"
+	// KubernetesPodNameLabel specify pod name label
```
Comment (mattjmcnaughton) [inline pkg/kubelet/types/labels.go:20]: Small nit :) I think we want to swap `specify` for `specifies a`.

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
Two small grammar nits from me, that would be great to clean up while we're working with this code.

Other than that looks good! Feel free to ping me if you make the change and I'll mark as lgtm.

File: pkg/kubelet/types/constants.go [hunk 2]
```diff
@@ -17,14 +17,17 @@ limitations under the License.
 package types
 
 const (
-	// system default DNS resolver configuration
+	// ResolvConfDefault define system default DNS resolver configuration
```
[the remark this answers is not in the document]
Comment (qw1mb0) [reply pkg/kubelet/types/constants.go]: fixed

File: pkg/kubelet/types/labels.go (hunk 1 above)
Comment (qw1mb0) [reply pkg/kubelet/types/labels.go]: fixed

**Comment by qw1mb0** [comment]
@mattjmcnaughton  I corrected the grammatical errors

File: pkg/kubelet/types/types.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func (t *Timestamp) GetString() string {
 	return t.time.Format(time.RFC3339Nano)
 }
 
-// A type to help sort container statuses based on container names.
+// SortedContainerStatuses a type to help sort container statuses based on container names.
```
Comment (mattjmcnaughton) [inline pkg/kubelet/types/types.go:64]: Nit: For this file, I think it could be `ConstantName is a...`

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
Sorry, one last thought and then looks good to me.

I'll mark lgtm once the final change is made.

File: pkg/kubelet/types/types.go (hunk 3 above)
Comment (qw1mb0) [reply pkg/kubelet/types/types.go]: > Nit: For this file, I think it could be `ConstantName is a...`

fixed

**Comment by qw1mb0** [comment]
@mattjmcnaughton ping

File: pkg/kubelet/types/types.go [hunk 4]
```diff
@@ -95,8 +95,8 @@ type Reservation struct {
 	Kubernetes v1.ResourceList
 }
 
-// A pod UID which has been translated/resolved to the representation known to kubelets.
+// ResolvedPodUID a pod UID which has been translated/resolved to the representation known to kubelets.
```
Comment (mattjmcnaughton) [inline pkg/kubelet/types/types.go:98]: Sorry, I wasn't clear in my original comment - we should also add the `is a` change for `ResolvedPodUID` and `MirrorPodUID` similar to `SortedContainerStatuses`.

File: pkg/kubelet/types/labels.go [hunk 5]
```diff
@@ -17,24 +17,32 @@ limitations under the License.
...
+	KubernetesPodNameLabel = "io.kubernetes.pod.name"
+	// KubernetesPodNamespaceLabel specifies a pod namespace label
+	KubernetesPodNamespaceLabel = "io.kubernetes.pod.namespace"
+	// KubernetesPodUIDLabel specifies a pod UID label
+	KubernetesPodUIDLabel = "io.kubernetes.pod.uid"
+	// KubernetesContainerNameLabel specifies a container name label
 	KubernetesContainerNameLabel = "io.kubernetes.container.name"
 )
 
+// GetContainerName return container name from labels
```
Comment (mattjmcnaughton) [inline pkg/kubelet/types/labels.go:30]: Nit: For all the functions in this file, I think we want `FuncName returns ...`.

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
Getting close!

Two final thoughts - I clarified one of my earlier comments - I hope it makes sense :) Please lmk if not!

File: pkg/kubelet/types/labels.go (hunk 5 above)
Comment (qw1mb0) [reply pkg/kubelet/types/labels.go:30]: fixed

File: pkg/kubelet/types/types.go (hunk 4 above)
Comment (qw1mb0) [reply pkg/kubelet/types/types.go:98]: fixed

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Thank you for these fixes!

**Review by SergeyKanzhelev — APPROVED** [review]

File: pkg/kubelet/types/pod_update.go [hunk 6]
```diff
@@ -25,42 +25,47 @@ import (
 )
 
 const (
-	ConfigSourceAnnotationKey    = "kubernetes.io/config.source"
-	ConfigMirrorAnnotationKey    = v1.MirrorPodAnnotationKey
+	// ConfigSourceAnnotationKey specifies a annotation key for config source
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/types/pod_update.go:28]: nit 

```suggestion
	// ConfigSourceAnnotationKey specifies an annotation key for config source
```

File: pkg/kubelet/types/pod_update.go [hunk 7]
```diff
@@ -25,42 +25,47 @@ import (
...
+	ConfigSourceAnnotationKey = "kubernetes.io/config.source"
+	// ConfigMirrorAnnotationKey specifies a annotation key for config mirror
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/types/pod_update.go:30]:
```suggestion
	// ConfigMirrorAnnotationKey specifies an annotation key for config mirror
```

File: pkg/kubelet/types/constants.go [hunk 8]
```diff
@@ -17,14 +17,17 @@ limitations under the License.
...
+	// DockerContainerRuntime defines a docker container runtime
 	DockerContainerRuntime = "docker"
+	// RemoteContainerRuntime defines a remote container runtime
 	RemoteContainerRuntime = "remote"
+)
 
-	// User visible keys for managing node allocatable enforcement on the node.
+// User visible keys for managing node allocatable enforcement on the node.
+const (
 	NodeAllocatableEnforcementKey = "pods"
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/types/constants.go:31]: curious why these `const` do not require a comment

**Comment by SergeyKanzhelev** [comment]
@qw1mb0 would you please rebase this PR?

**Comment by SergeyKanzhelev** [comment]
new attempt is here: https://github.com/kubernetes/kubernetes/pull/97374

/close

**Pull request closed without merging**